apps/wpcom-block-editor/src/wpcom/features/tracking.js

```diff
@@ -114,12 +114,21 @@ const getBlocksTracker = ( eventName ) => ( blockIds ) => {
  * Track block insertion.
  *
  * @param {object|Array} blocks block instance object or an array of such objects
+ * @param {Array} args additional insertBlocks data e.g. metadata containing pattern name.
  * @returns {void}
  */
-const trackBlockInsertion = ( blocks ) => {
+const trackBlockInsertion = ( blocks, ...args ) => {
+	const patternName =
+		4 === args.length && isPlainObject( args[ 3 ] ) ? args[ 3 ].patternName : undefined;
```

Comment:
fyi, you don't need to use yoda conditions in js code, it is only enforced in wp php code - https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#yoda-conditions - you don't need to change this instance, as it is allowed, just not enforced, just pointing out in case you didn't know this

Reply:
Is there a way we can protect against the position of the meta argument moving in the future? Do you think that is worth doing?

Reply:
we could use:
```javascript
	let patternName;
	args.forEach( arg => {
		if ( arg?.patternName ) {
			patternName = arg.patternName;
		}
	} );
```

Reply:
Thanks, yoda conditions have become a habit. I'll add the suggestion to make this more robust in the face of the arguments moving around. It was a bit of a smell that needed addressing.